### PR TITLE
Event emissions

### LINKED ISSUE #46: Discussion: Event changes
## REGISTRY

`_Application`: add `appEndDate`

`_Challenge`: remove `deposit`, add `challenger`, change `pollID` to `challengeID`

`_NewListingWhitelisted`: change to `_ApplicationWhitelisted`

Add: (#35 & #45) 
- `_ListingWithdrawn(bytes32 listingHash);`
- `_TouchAndRemoved(bytes32 listingHash);`

Current Implementation:
```sol
event _Application(bytes32 listingHash, uint deposit, string data);
event _Challenge(bytes32 listingHash, uint deposit, uint pollID, string data);
event _Deposit(bytes32 listingHash, uint added, uint newTotal);
event _Withdrawal(bytes32 listingHash, uint withdrew, uint newTotal);
event _NewListingWhitelisted(bytes32 listingHash);
event _ApplicationRemoved(bytes32 listingHash);
event _ListingRemoved(bytes32 listingHash);
event _ChallengeFailed(uint challengeID);
event _ChallengeSucceeded(uint challengeID);
event _RewardClaimed(address voter, uint challengeID, uint reward);
```

Proposed implementation:
```sol
* event _Application(bytes32 listingHash, uint deposit, uint appEndDate, string data);
* event _Challenge(bytes32 listingHash, address challenger, uint challengeID, string data);
  event _Deposit(bytes32 listingHash, uint added, uint newTotal);
  event _Withdrawal(bytes32 listingHash, uint withdrew, uint newTotal);
* event _ApplicationWhitelisted(bytes32 listingHash);
  event _ApplicationRemoved(bytes32 listingHash);
  event _ListingRemoved(bytes32 listingHash);
+ event _ListingWithdrawn(bytes32 listingHash);
+ event _TouchAndRemoved(bytes32 listingHash);
  event _ChallengeFailed(uint challengeID);
  event _ChallengeSucceeded(uint challengeID);
  event _RewardClaimed(address voter, uint challengeID, uint reward);
```

---

## PARAMETERIZER

Add: (#34)
- `_RewardClaimed(msg.sender, _challengeID, reward);`

Add: (#33)
- `_ProposalAccepted(bytes32 propID);` 
- `_ProposalUnprocessed(bytes32 propID);`
- `_ChallengeSucceeded(uint challengeID);`
- `_ChallengeFailed(uint challengeID);`

Current:
```sol
event _ReparameterizationProposal(address proposer, string name, uint value, bytes32 propID);
event _NewChallenge(address challenger, bytes32 propID, uint pollID);
```

Proposed:
```sol
  event _ReparameterizationProposal(address proposer, string name, uint value, bytes32 propID);
  event _NewChallenge(address challenger, bytes32 propID, uint pollID);
+ event _ProposalAccepted(bytes32 propID);
+ event _ProposalUnprocessed(bytes32 propID);
+ event _ChallengeSucceeded(uint challengeID);
+ event _ChallengeFailed(uint challengeID);
+ event _RewardClaimed(address voter, uint challengeID, uint reward);
```

---

## PLCRVOTING

`PollCreated`: change args to `commitEndDate`, `revealEndDate`

Add: underscores for convention

Experimental:
`VoteRevealed`: change to `(voter, pollID, votesFor, votesAgainst)`

Current:
```sol
event VoteCommitted(address voter, uint pollID, uint numTokens);
event VoteRevealed(address voter, uint pollID, uint numTokens, uint choice);
event PollCreated(uint voteQuorum, uint commitDuration, uint revealDuration, uint pollID);
event VotingRightsGranted(address voter, uint numTokens);
event VotingRightsWithdrawn(address voter, uint numTokens);
```

Proposed:
```sol
_  event _VoteCommitted(address voter, uint pollID, uint numTokens);
*  event _VoteRevealed(address voter, uint pollID, uint votesFor, uint votesAgainst);
*  event _PollCreated(uint voteQuorum, uint commitEndDate, uint revealEndDate, uint pollID);
_  event _VotingRightsGranted(address voter, uint numTokens);
_  event _VotingRightsWithdrawn(address voter, uint numTokens);
```

### PR DESCRIPTION
**Pull request by kangarang**
### Description

Closes #46

Adds `indexed` to certain desirable event arguments. This enables UIs to filter and search for specific values. This PR also adds additional arguments to certain events to trigger UI changes more conveniently.

> Up to three parameters can receive the attribute indexed which will cause the respective arguments to be searched for: It is possible to filter for specific values of indexed arguments in the user interface.

[solidity events docs](http://solidity.readthedocs.io/en/latest/contracts.html#events)

### Changes

* add `indexed` to relevant event emission args
* `_ChallengeFailed` and `_ChallengeSucceeded`: add `rewardPool` and `totalTokens` args
* `_NewChallenge`: add `propID`, `commitEndDate` and `revealEndDate`
* `_NewChallenge`: change `pollID` to `challengeID`
* adjust tests accordingly

### CODE REVIEW

**Comment by kangarang** [comment]
@eccheung4

**Comment by eccheung4** [comment]
@kangarang looks good! just wondering if there should also be a _ApplicationExpired event in Registry.sol?
Are we missing a case in updateStatus in Registry.sol like processProposal in Parameterizer.sol?
`} else if (now > prop.processBy) {`
An application always reverted if updateStatus isn't called before the processBy?

**Comment by eccheung4** [comment]
Proposing we add back `choice` to
`event _VoteRevealed(uint pollID, uint numTokens, uint votesFor, uint votesAgainst);`
so it becomes
`event _VoteRevealed(uint pollID, uint numTokens, uint choice, uint votesFor, uint votesAgainst);`

Without it, we have to do a bit more digging to figure out user's vote

**Comment by skmgoldin** [comment]
@eccheung4 the Registry doesn't have any `processBy` logic at all. Arguably it should, but it does not.

**Comment by skmgoldin** [comment]
"Loitering" attacks on the Parameterizer could be quite bad, but a loitering attack on the registry just means it contains one bad listing until someone challenges it.

**Comment by kangarang** [comment]
@eccheung4  for registry applications, it's not so much a big deal if the application doesn't get processed since the only entity that's affected by that scenario is the applicant. but a reparameterization proposal affects the entire network, so the processBy date ensures that reparameterizations are followed up promptly

**Comment by kangarang** [comment]
adding in `choice` right now

**Comment by skmgoldin** [comment]
Lets review it in code review tomorrow.

File: contracts/Parameterizer.sol [hunk 1]
```diff
@@ -152,7 +152,7 @@ contract Parameterizer {
 
     require(token.transferFrom(msg.sender, this, deposit)); // escrow tokens (deposit amt)
 
-    _ReparameterizationProposal(_name, _value, propID);
+    _ReparameterizationProposal(_name, _value, propID, deposit, now.add(get("pApplyStageLen")));
```
Comment (kangarang) [inline contracts/Parameterizer.sol]: `proposals[propID].appExpiry`

File: contracts/Parameterizer.sol [hunk 2]
```diff
@@ -186,7 +186,10 @@ contract Parameterizer {
     //take tokens from challenger
     require(token.transferFrom(msg.sender, this, deposit));
 
-    _NewChallenge(_propID, pollID);
+    uint commitEndDate = block.timestamp.add(get("pCommitStageLen"));
+    uint revealEndDate = commitEndDate.add(get("pRevealStageLen"));
+
+    _NewChallenge(_propID, pollID, commitEndDate, revealEndDate);
```
Comment (kangarang) [inline contracts/Parameterizer.sol]: get these variables directly from plcrvoting pollMap[pollID]

**Pull request merged**